libs/qlik/src/dashboard/vl-qlik-dashboard.component.ts

```diff
@@ -0,0 +1,315 @@
+        };
+    }
+
+    private renderVisualRow(visualRow: DashboardVisualization[]) {
+        return visualRow.map((v) => {
+            if (!Array.isArray(v)) {
+                return {
+                    size: v.colSize || 1,
+                    maxSize: visualRow.length,
+                    template: this.renderVisual(v),
+                };
+            } else {
+                return {
+                    size: 1,
+                    maxSize: visualRow.length,
+                    template: renderStack(...this.renderVisualRow(v)),
+                };
+            }
+        });
+    }
+
+    private renderVisual(v: DashboardVisualization) {
+        return html`
+            <label is="vl-form-message" for="visual-${v.id}" class="visual-label-100-${v['align-label'] || 'left'}"
+                >${v.label}</label
+            >
+            <div style="min-height: ${v.height}">
+                <vl-qlik-visual
+                    class="visual-100-${v['align-visual'] || 'left'}"
+                    qlik-id="${v.id}"
+                    type="${v.type}"
+                    height="${v.height}"
+                    width="${v.width}"
+                    .stardust="${this.stardust}"
+                    .properties="${v.properties}"
+                    .options="${v.options}"
+                    @visual-changed="${this.visualChanged}"
+                    additional-style="position: absolute; width:95%; bottom:0"
+                ></vl-qlik-visual>
+            </div>
+        `;
+    }
+
+    private async bindFilters() {
+        const filterVs = (
+            await Promise.all(
+                this.filters.map(async (filter) => {
+                    const filterValues = await this.connection.getFilterValues(filter.filter.name, true);
+                    bindVlSelect({
+                        component: queryById(this)(filter.id),
+                        choices: filterValues.map((filterValue) => {
+                            return {
+                                label: filterValue.label,
+                                value: filterValue.label,
+                                disabled: filterValue.state === 'excluded',
+                            };
+                        }),
+                        selectedChoices: this.selected[this.fieldValue(filter)]
+                            ? this.selected[this.fieldValue(filter)].selections
+                            : [],
+                    });
+                    return { filtername: filter.filter.name, values: filterValues, filterid: filter.id };
+                })
+            )
+        ).reduce((result, item) => {
+            result[item.filtername] = item;
+            return result;
+        }, {});
+
+        Object.keys(filterVs).forEach((f) => {
+            queryBySelectorAll(this)(`div[id^="choices-${filterVs[f].filterid}-item"]`).forEach((c) => {
```

Comment:
kan je ofwel `this.getRootNode()` gebruiken (zie eerdere comment) of indien niet, wel zorgen dat het resultaat die uit deze functie komt getypt is (nu is geeft dit `any` terug)